### ISSUE TITLE
High error rate for KITTI 2015 Training dataset

### ISSUE THREAD

**Issue by Diksha-Moolchandani** [lmb-freiburg/flownet2#223]
I ran the flownet2-kitti model provided in the repo. The dataset used was KITTI 2015 training dataset with left images in the image_2/ folder, right images in the image_3/ folder, and the ground truth disparity in the disp_occ_0/ folder. I have modified run-flownet-many.py with the Kitti dataloader.

I am getting an average error of 0.59 (averaged across the 200 images in the dataset).

The error is calculated using compareDisp(ground_truth, blob). This function is called after line [98](https://github.com/lmb-freiburg/flownet2/blob/b92e198b56b0e52e1ba0a5a98dc0e39fa5ae70cc/scripts/run-flownet.py#L98) in run-flownet.py.
The function is described below.

**def compareDisp(disp_L, flow):**
    
    disp = flow[:,:,0:1]
    [height, width, channel] = flow.shape
    disp = np.absolute(np.reshape(disp, (height, width)))
    gt = disp_L
    disp = torch.tensor(disp)
    disp = torch.unsqueeze(disp,0)
    mask = gt > 0
    errmap = torch.abs(disp - gt)
    err3 = ((errmap[mask] > 3.) & (errmap[mask] / gt[mask] > 0.05)).sum()
    print((err3.float() / mask.sum().float()).item())
    return (err3.float() / mask.sum().float()).item()

The error seems very large. Can anyone help me with this? Similar function is used in [ISM algorithm (MICRO paper)](https://github.com/horizon-research/ism-algorithm/blob/be73b84630ac768a085c9f159ba33cb5404001b5/stereo_script/flownet/flownet_with_ism.py#L114) and [AnyNet code](https://github.com/mileyan/AnyNet/blob/83badc75af596e76edce67e4917ba4fbacbb7c1d/finetune.py#L209)
I can share the modified python file, if needed.

**Comment by nikolausmayer** [comment]
If I read this correctly, the error is measured in pixels, right? 0.59 isn't a large pixel error, in fact it's probably very good, considering  the fact that FlowNet2 is _not_ specialized for disparity.
Did you check the results visually, e.g. by warping the right image onto the left and looking at the difference to the original left image?
